openfisca_nsw_safeguard/variables/ESS_PDRS_Estimator/RF2/certificate_estimation/RF2_ESC_variables.py

```diff
@@ -36,67 +36,135 @@ class RF2_total_display_area(Variable):
+          parameters(period).PDRS.refrigerated_cabinets.table_RF2_1['adjustment_factor'][product_class][duty_type]
+        ]
+      )
+      return af
     
     
 class RF2_baseline_EEI(Variable):
-  value_type = float
-  entity = Building
-  definition_period = ETERNITY
-  label = "Baseline EEI"
+    value_type = float
+    entity = Building
+    definition_period = ETERNITY
+    label = "Baseline EEI"
   
-  def formula(buildings, period, parameters):
-    product_class = buildings('RF2_product_class', period)
-    duty_type = buildings('RF2_duty_class', period)
-    replacement_activity = buildings('RF2_replacement_activity', period)
-
-    baseline_EEI = np.select(
-      [ 
-        replacement_activity,
-        np.logical_not(replacement_activity)
-      ],
-      [ 
-        parameters(period).PDRS.refrigerated_cabinets.table_RF2_1['baseline_EEI'][product_class][duty_type],
-        parameters(period).ESS.HEAB.table_F1_1_1['baseline_EEI'][product_class][duty_type]
-      ]
-    )    
-    return baseline_EEI
+    def formula(buildings, period, parameters):
+      product_class = buildings('RF2_product_class', period)
+      duty_type = buildings('RF2_duty_class', period)
+      replacement_activity = buildings('RF2_replacement_activity', period)
+
+      baseline_EEI = np.select(
+        [ 
+          replacement_activity,
+          np.logical_not(replacement_activity)
+        ],
+        [
+          parameters(period).PDRS.refrigerated_cabinets.table_RF2_1['baseline_EEI'][product_class][duty_type],
+          parameters(period).ESS.HEAB.table_F1_1_1['baseline_EEI'][product_class][duty_type]
+        ]
+      )
+      return baseline_EEI
 
 
 class RF2_product_EEI(Variable):
-  value_type = float
-  entity = Building
-  definition_period = ETERNITY
-  metadata = {
-    'variable-type' : 'user-input',
-    'label' : 'Product EEI',
-    'display_question' : 'Energy Efficiency Index of the replacement refrigerated cabinet model as recorded in the GEMS Registry',
-    'sorting' : 8
-  }
-  
-  
+    value_type = float
+    entity = Building
+    definition_period = ETERNITY
+    metadata = {
+      'variable-type' : 'user-input',
+      'label' : 'Product EEI',
+      'display_question' : 'Energy Efficiency Index of the replacement refrigerated cabinet model as recorded in the GEMS Registry',
+      'sorting' : 8
+    }
+
+
+class RF2_product_EEI_ESC_replacement_eligibility(Variable):
+    value_type = bool
+    entity = Building
+    definition_period = ETERNITY
+
+    def formula(building, period, parameters):
+      product_EEI = building('RF2_product_EEI', period)
+      product_class_5 = building('RF2_product_class', period)
+        
+      product_EEI_to_check_ESC = np.select(
+          [
+              (product_EEI < 51),
+              (product_EEI >= 51) * (product_EEI < 81) * (product_class_5 != RF2ProductClass.product_class_five),
```

Comment:
this should be

np.logical_not(product_class_5 == RF2ProductClass.product_class_five)